content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,361 +36,583 @@ are covered here as well as the `System.Diagnostics` API.
 If you prefer to use OpenTelemetry APIs instead of `System.Diagnostics` APIs,
 you can refer to the [OpenTelemetry API Shim docs for tracing](../shim).
 
-## Traces
+## Example app preparation {#example-app}
 
-### Initializing tracing
+This page uses a modified version of the example app from
+[Getting Started](/docs/languages/net/getting-started/) to help you learn
+about manual instrumentation.
 
-There are two main ways to initialize [tracing](/docs/concepts/signals/traces/),
-depending on whether you're using a console app or something that's ASP.NET
-Core-based.
+You don't have to use the example app: if you want to instrument your own app or
+library, follow the instructions here to adapt the process to your own code.
 
-#### Console app
+### Dependencies {#example-app-dependencies}
 
-To start tracing in a console app, you need to create a tracer provider.
+- [.NET SDK](https://dotnet.microsoft.com/download/dotnet) 6+
 
-First, ensure that you have the right packages:
+### Create and launch an HTTP Server
 
-```sh
-dotnet add package OpenTelemetry
-dotnet add package OpenTelemetry.Exporter.Console
+To highlight the difference between instrumenting a library and a standalone
+app, split out the dice rolling into a library file, which then will be
+imported as a dependency by the app file.
+
+Create the library file named `Dice.cs` and add the following code to it:
+
+```csharp
+/*Dice.cs*/
+namespace otel
+{
+    public class Dice
+    {
+        private int min;
+        private int max;
+
+        public Dice(int min, int max)
+        {
+            this.min = min;
+            this.max = max;
+        }
+
+        public List<int> rollTheDice(int rolls)
+        {
+            List<int> results = new List<int>();
+
+            for (int i = 0; i < rolls; i++)
+            {
+                results.Add(rollOnce());
+            }
+            
+            return results;
+        }
+
+        private int rollOnce()
+        {
+            return Random.Shared.Next(min, max + 1);
+        }
+    }
+}
 ```
 
-And then use code like this at the beginning of your program, during any
-important startup operations.
+Create the app file `DiceController.cs` and add the following code to it:
 
 ```csharp
-using OpenTelemetry;
-using OpenTelemetry.Resources;
-using OpenTelemetry.Trace;
+/*DiceController.cs*/
+using Microsoft.AspNetCore.Mvc;
+using System.Net;
 
-// ...
+namespace otel
```

Comment:
Do we need a namespace here? (what value does it provide to the user)
If we need a namespace, consider following the .NET convention and use upper camel case.

Reply:
consider https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces

It reduces nesting.